errors.go

```diff
@@ -109,6 +124,40 @@ func sprintf(format string, args ...interface{}) string {
 	return fmt.Sprintf(format, args...)
 }
 
+// NewBatch creates a new BatchError.
+func NewBatch() BatchError {
+	return &batcherrtype{}
+}
+
+//Append appends the error
+func (b *batcherrtype) Append(err error) {
+	b.errors = append(b.errors, err)
+}
+
+//Errors returns the errors
+func (b *batcherrtype) Errors() []error {
+	return b.errors
+}
+
+//IsEmpty returns if there is any error
+func (b *batcherrtype) IsEmpty() bool {
+	return len(b.errors) == 0
+}
+
+// Error implements the standard library error interface.
+func (b *batcherrtype) Error() string {
+	if b.IsEmpty() {
+		return ""
+	}
+
+	message := ""
+	for _, r := range b.Errors() {
+		message += r.Error() + ";"
```

Comment:
Please be more consistent with error concatenation. `Wrap` uses `"."` to concatenate errors.